### PR TITLE
Change the ListView on the MainActivity, closes #10

### LINKED ISSUE #10: Screen with list with contacts
- Parent issues: #1 
- Relevant issues: -
- Estimated effort: 2

* * *

## User story
As a user I want to have a screen with a list of names of the people of whom I have a public key so I can see my contacts.

## What
This issue demands to have the screen with a list of the contacts synchronized with the database

## Why
This issue is needed for the project because the user must be able to see which contacts he has

## How
This feature can be added by synchronizing the database with the contact screen

## Definition of done
This task can be considered done when

- The contact screen always reflects the current state of the database

### PR DESCRIPTION
**Pull request by clenrock**
- Relevant Issues: #10 

* * *

## What
A nicer contacts list with their own Adapter, so changing it will be easier in the future. This will also look in the database for the contacts instead of getting it from an ArrayList with dummy data.

## Why
It gives a nicer user experience and it had to be connected to the database.

## How
When you run the app and go to the MainActivity you can see the new list.

## Notes
In the future we should try to find a solution for the grey part below the list, when you don't have a lot of contacts yet.

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java [hunk 1]
```diff
@@ -52,21 +62,100 @@ public void onClick(final View view) {
...
+class ContactListAdapter extends ArrayAdapter<Contact> {
+
+    private final static int MAX_SNIPPET_SIZE = 30;
+    /**
+     * Create and initialize a ContactListAdapter.
+     *
+     * @param context the Context where the ListView is created
+     * @param contacts the list with contacts
+     */
+    ContactListAdapter(final Context context, final List<Contact> contacts) {
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: Can you make this class final, assuming you don't want to extend this class later on?

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java [hunk 2]
```diff
@@ -52,21 +62,100 @@ public void onClick(final View view) {
...
+        super(context, 0, contacts);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public View getView(final int position, final View convertView, final ViewGroup parent) {
+
+        View v = convertView;
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: final

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java [hunk 3]
```diff
@@ -52,21 +62,100 @@ public void onClick(final View view) {
...
+            final TextView name = (TextView) v.findViewById(R.id.name);
+            final TextView pubKey = (TextView) v.findViewById(R.id.pubKeySnippet);
+
+            if (name != null) {
+                name.setText(contact.getName());
+            }
+
+            if (pubKey != null) {
+                final String publicKey = contact.getPublicKey().getKey();
+                if(publicKey.length() > MAX_SNIPPET_SIZE) {
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: Ctrl-alt-shift-L (no space between if and round bracket)

File: app/src/main/res/layout/contact_list_entry.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            android:layout_height="wrap_content"
+            android:text="Name"
+            android:textSize="@dimen/contacts_list_entry_text_name_size" />
+
+        <TextView
+            android:id="@+id/pubKeySnippet"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:gravity="center_vertical"
+            android:text="Public Key snippet" />
```
Comment (jverbraeken) [inline app/src/main/res/layout/contact_list_entry.xml]: Can you put this in strings.xml?

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java (hunk 2 above)
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: This is changed later on, and shouldn't be final

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java (hunk 1 above)
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: Good point

**Pull request merged**